rhoas/kafkas/resource_kafka.go

```diff
@@ -249,9 +254,85 @@ func kafkaCreate(ctx context.Context, d *schema.ResourceData, m interface{}) dia
 		return diag.FromErr(err)
 	}
 
+	// now that kafka is created define the acl
+	err = createACLForKafka(ctx, api, d, &kafka)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
 	return diags
 }
 
+func createACLForKafka(ctx context.Context, api rhoasAPI.Clients, d *schema.ResourceData, kafka *kafkamgmtclient.KafkaRequest) error {
+
+	acl, ok := d.Get("acl").([]interface{})
+	if !ok {
+		return errors.Errorf("No acl defined in the kafka resource")
+	}
+
+	for i := 0; i < len(acl); i++ {
+		element, ok := acl[i].(map[string]interface{})
+		if !ok {
+			return errors.Errorf("Cannot cast contents of acl to a map[string]interface{}")
+		}
+
+		principal, ok := element["principal"].(string)
+		if !ok {
+			return errors.Errorf("There was a problem getting the principal value in the kafka acl")
+		}
+
+		// required for api, the user id, service account id or * works
+		// when appended to User:
+		principal = "User:" + principal
```

Comment:
is there any validation to check if the user passes the principal in as `User:blah` or `user:blah` or `user: blah`, you get my point, in the TF script?
in those cases wouldn't the principal come out as `User:User:blah` ?

Reply:
my preference here would be to have a little validation around the prepending of `User:` here but that is just my own opinion

Reply:
I would normally agree but... then doing what I am in this pr not impossible but a mess, I am using a service account client id as the principle but I would need to prepend it with User:, this afaik is possible but a pain in terraform

Reply:
now i have realized doing this changing of values can cause issues with terraform syncing after create, i avoided it in this pr in a hacky way but something to think about for the future.